### ISSUE TITLE
bug in gzmat conversion in version 2.2.1

### ISSUE THREAD

**Issue by openbabel-bot** [openbabel/openbabel#872]
There is a problem when converting a gzmat file to an other file format and
in the other way round.
When converting to a gzmat file, the program put sometimes nan
When converting from a gzmat, the program put all the y values to 0 and
even sometimes all the values bellow the 4 atoms are "nan".

This problem is arising since the new version 2.2.1

As an example, i took a first file a.xyz and convert it into b.gzmat: babel
-ixyz a.xyz -ogzmat b.gzmat. Four values namely d76, d116, d126 and d156
are "nan"
If i fixed these values to 180 degree,
Then convert back the gzmat into xyz: babel -igzmat c.xyz -oxyz c.xyz, i
have zero for y values and some nan. It does not correspond at all to the
init file a.xyz except for the three first atoms

I have try the same procedure with the 2.2.0 version.
I join the file as vers220.tar file.
The a.xyz file is the same.
The b2.gzmat is the new version of the file done with babel 2.2.0 that is
in fact the same that with version 2.2.1 that means that the four "nan" are
also found with the older version.
If i fixed these values to 180 degrees (b2corr.gzmat) and then do the
conversion back to xyz: c2.xyz, the result is good. We find back the same
geometry as for a.xyz with some rotation: permutation of x and z and y->-y
that correspond to a rotation of 90 degree along y axis followed by a
rotation of 180 degree along the new x axis.

In summary: going from xyz -> gzmat. bug in both version: some "nan"
values
going from gzmat -> xyz, new bug: some "nan" and y
values are all zero.

Regards

Vincent

Reported by: vliegeois
